### PR TITLE
Make Nano use json_encode.

### PR DESCRIPTION
**Pull request by PJB3005**
This makes it possible to send lists of strings.

[system] [ui] [performance]

### CODE REVIEW

File: code/modules/nano/nanoui.dm [hunk 1]
```diff
@@ -502,7 +502,7 @@ nanoui is used to open and update nano browser uis
 
 //	to_chat(user, data ? list2json(data) : "null")// used for debugging
```
Comment (DamianX) [inline code/modules/nano/nanoui.dm]: You can remove the ternary operator and fix this one as well

**Comment by Exxion** [comment]
I wonder what will happen with infinite/NaN values on interfaces
Not that it's an especially big deal since under no normal circumstances does an interface try to display an infinite value

EDIT: The answer, shockingly, is that it doesn't work
There's probably a way to fix this that's less half-assed than what I did before though

**Comment by PJB3005** [comment]
@Exxion is that actually something that got broken by my code or nah?

**Comment by brndd** [comment]
There was also apparently a bug where the custom json parser needlessly html_encodes apostrophes, and I think it's what's causing my State Laws rewrite to convert apostrophes into backticks. I think this is going to fix that too.

**Comment by PJB3005** [comment]
@brndd correct.

**Comment by Exxion** [comment]
@PJB3005 It was """fixed""" by a change to the JSON writer before, so yes

**Comment by PJB3005** [comment]
Burneddi is gonna merge this PR into his own branch for something so merge this with a merge commit, not a squash merge.

**Comment by brndd** [comment]
Yeah I need this because being only able to use associative lists in nanoui is fucking frying my brain. Also the old json parser replaces every single apostrophe with a backtick 😩

**Pull request merged**